channeldb/graph_test.go

```diff
@@ -53,27 +51,18 @@ var (
 	testPub = route.Vertex{2, 202, 4}
 )
 
-// MakeTestGraph creates a new instance of the ChannelGraph for testing
-// purposes. A callback which cleans up the created temporary directories is
-// also returned and intended to be executed after the test completes.
-func MakeTestGraph(modifiers ...OptionModifier) (*ChannelGraph, func(), error) {
-	// First, create a temporary directory to be used for the duration of
-	// this test.
-	tempDirName, err := ioutil.TempDir("", "channelgraph")
-	if err != nil {
-		return nil, nil, err
-	}
-
+// MakeTestGraph creates a new instance of the ChannelGraph for testing purposes.
+func MakeTestGraph(t testing.TB, modifiers ...OptionModifier) (*ChannelGraph, error) {
 	opts := DefaultOptions()
 	for _, modifier := range modifiers {
 		modifier(&opts)
 	}
 
 	// Next, create channelgraph for the first time.
-	backend, backendCleanup, err := kvdb.GetTestBackend(tempDirName, "cgr")
+	backend, backendCleanup, err := kvdb.GetTestBackend(t.TempDir(), "cgr")
 	if err != nil {
-		backendCleanup()
-		return nil, nil, err
+		t.Cleanup(backendCleanup)
```

Comment:
I think the cleanup function can be called directly here as we're going to abort anyway. Same in the error case on L75.

Reply:
Resolved in 889bcd4727c6600cafd9bf9b8994db2a2c68951f